Issue building torch-android

Hi,

I'm trying to compile torch-android and get some issues.

First I installed Torch using the instructions given [here](http://torch.ch/docs/getting-started.html#_).

Then I followed the instructions given for torch-android [here](https://github.com/soumith/torch-android/blob/master/README.md).

I set ANDROID_NDK_TOOLCHAIN_ROOT to my android ndk path, as specified in the instructions:

```
$ echo $ANDROID_NDK_TOOLCHAIN_ROOT 
/home/jul/Development/android-ndk-r10d
```

But when running `sh build.sh` I get:

`Torch-7 not found on system. Installing`

probably because the install script recommended installed Torch locally. If I remove it and let your script install it, but I get:

```
$sh build.sh 
Torch-7 not found on system. Installing.
build.sh: 16: build.sh: [[: not found
build.sh: 18: build.sh: [[: not found
Android NDK found at: 
rm: cannot remove ‘build/CMakeCache.txt’: Permission denied
rm: cannot remove ‘build/CMakeFiles/cmake.check_cache’: Permission denied
CMake Warning at cmake/android.toolchain.cmake:404 (message):
  Using value of obsolete variable ANDROID_NDK_TOOLCHAIN_ROOT as initial
  value for ANDROID_STANDALONE_TOOLCHAIN.  Please note, that
  ANDROID_NDK_TOOLCHAIN_ROOT can be completely removed in future versions of
  the toolchain.
Call Stack (most recent call first):
  cmake/android.toolchain.cmake:506 (__INIT_VARIABLE)
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Error at cmake/android.toolchain.cmake:429 (FILE):
  file STRINGS file
  "/home/jul/Development/android-ndk-r10d/sysroot/usr/include/android/api-level.h"
  cannot be read.
Call Stack (most recent call first):
  cmake/android.toolchain.cmake:619 (__DETECT_NATIVE_API_LEVEL)
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Warning at cmake/android.toolchain.cmake:444 (message):
  Could not determine machine name for compiler from
  /home/jul/Development/android-ndk-r10d
Call Stack (most recent call first):
  cmake/android.toolchain.cmake:622 (__DETECT_TOOLCHAIN_MACHINE_NAME)
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Error at cmake/android.toolchain.cmake:624 (message):
  Could not determine machine name of your toolchain.  Probably your Android
  standalone toolchain is broken.
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCCompiler.cmake
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCXXCompiler.cmake
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
CMake Error: Cannot open file for write: /home/jul/Development/torch-android/src/build/CMakeCache.txt.tmp
CMake Error: : System Error: Permission denied
CMake Error: Unable to open cache file for save. /home/jul/Development/torch-android/src/build/CMakeCache.txt
CMake Error: : System Error: Permission denied
CMake error. Exiting.
```


and if I run it with sudo I get

```
$ sudo sh build.sh 
[sudo] password for jul: 
Torch-7 not found on system. Installing.
build.sh: 16: build.sh: [[: not found
build.sh: 18: build.sh: [[: not found
Android NDK found at: 
CMake Error at cmake/android.toolchain.cmake:562 (message):
  Could not find neither Android NDK nor Android standalone toolchain.

      You should either set an environment variable:
        export ANDROID_NDK=~/my-android-ndk
      or
        export ANDROID_STANDALONE_TOOLCHAIN=~/my-android-toolchain
      or put the toolchain or NDK in the default path:
        sudo ln -s ~/my-android-ndk /opt/android-ndk
        sudo ln -s ~/my-android-toolchain /opt/android-toolchain
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCCompiler.cmake
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCXXCompiler.cmake
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
CMake error. Exiting.
```

Can you help?

Thank you

Same error after delete/reclone. It seems that your build.sh script is looking for "torch-lua" which is not installed by the recommended installation for Torch given [here](http://torch.ch/docs/getting-started.html#_). Here's what is installed:

```
$ ll torch/install/bin/
total 732
-rwxrwxr-x 1 jul jul    120 févr. 17 12:25 itorch
-rwxrwxr-x 1 jul jul    662 févr. 17 12:25 itorch_launcher
-rwxr-xr-x 1 jul jul    553 févr. 17 12:22 json2lua
-rwxr-xr-x 1 jul jul    553 févr. 17 12:22 lua2json
-rwxr-xr-x 1 jul jul 692622 févr. 17 12:22 luajit
-rwxr-xr-x 1 jul jul    996 févr. 17 12:22 luarocks
-rwxr-xr-x 1 jul jul    627 févr. 17 12:22 luarocks-admin
-rwxr-xr-x 1 jul jul    542 févr. 17 12:22 mdcat
-rwxr-xr-x 1 jul jul  18766 févr. 17 12:24 qlua
-rwxr-xr-x 1 jul jul    535 févr. 17 12:24 th
```

Here's the error (same as before):

```
$ sudo sh build.sh 
Torch-7 not found on system. Installing.
build.sh: 16: build.sh: [[: not found
build.sh: 18: build.sh: [[: not found
Android NDK found at: 
CMake Error at cmake/android.toolchain.cmake:562 (message):
  Could not find neither Android NDK nor Android standalone toolchain.

      You should either set an environment variable:
        export ANDROID_NDK=~/my-android-ndk
      or
        export ANDROID_STANDALONE_TOOLCHAIN=~/my-android-toolchain
      or put the toolchain or NDK in the default path:
        sudo ln -s ~/my-android-ndk /opt/android-ndk
        sudo ln -s ~/my-android-toolchain /opt/android-toolchain
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/CMakeDetermineSystem.cmake:90 (include)
  CMakeLists.txt:2 (PROJECT)


CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_C_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCCompiler.cmake
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER_ENV_VAR
CMake Error: Error required internal CMake variable not set, cmake may be not be built correctly.
Missing variable is:
CMAKE_CXX_COMPILER
CMake Error: Could not find cmake module file: /home/jul/Development/torch-android/src/build/CMakeFiles/2.8.12.2/CMakeCXXCompiler.cmake
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
CMake error. Exiting.
```

--- a/build-neon.sh
+++ b/build-neon.sh
@@ -26,7 +26,7 @@ cd src
 rm -rf build
 mkdir -p build
 cd build
-cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/android.toolchain.cmake .. -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DANDROID_STL=none -DARM_TARGETS="armeabi-v7a with NEON" -DNEON_FOUND=true 
+cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/android.toolchain.cmake .. -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DANDROID_STL=none -DARM_TARGETS="armeabi-v7a with NEON" -DNEON_FOUND=true -DLIBRARY_OUTPUT_PATH_ROOT=..
 CMAKERET=$?
 if [ $CMAKERET -ne 0 ]; then
  echo "CMake error. Exiting."

--- a/build.sh
+++ b/build.sh
@@ -26,7 +26,7 @@ cd src
 rm -rf build
 mkdir -p build
 cd build
-cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/android.toolchain.cmake .. -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DANDROID_STL=none
+cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/android.toolchain.cmake .. -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DANDROID_STL=none -DLIBRARY_OUTPUT_PATH_ROOT=..
 CMAKERET=$?
 if [ $CMAKERET -ne 0 ]; then
  echo "CMake error. Exiting."

